runtime/src/main/java/org/corfudb/runtime/collections/Table.java

```diff
@@ -372,6 +373,30 @@ public void close() throws Exception {
         corfuTable.close();
     }
 
+    public boolean matchesParams(@Nonnull final Class<K> kClass,
+                                 @Nonnull final Class<V> vClass,
+                                 @Nullable final Class<M> mClass,
+                                 @Nonnull final TableOptions tableOptions) {
+        if (this.keyClass != kClass) {
+            return false;
+        }
+        if (this.valueClass != vClass) {
+            return false;
+        }
+        if (this.metadataClass != mClass) {
```

Comment:
this can throw an NPE if metadata class is null

Reply:
it would just compare null with another value right - why would it throw a NPE?